### ISSUE TITLE
RFC: Get distribution specific package source directory

### ISSUE THREAD

**Issue by harish-24** [avocado-framework/avocado#2283]
Idea is to provide a distribution-specific "ready-to-build" source of a given package.

Use-case:
Tests makes use of the build-directory and run selftest (if required) for any distribution-specific package. As each source may be built in a different way, a "ready-to-build" source path would be ideal 
to return. 

Prerequisite: Enable source package repository and Optional package repository for installing build-dependencies.
Input: package-name, ready-to-build source path
Output: True if available, else False

For implementation point of view, it would be a function in SoftwareManager handling various distributions.

Would be helpful to have your thoughts on both the approach and implementation.

**Comment by lmr** [comment]
I'm not sure whether most distributions do have source packages (in the sense Red Hat does), although I suppose the implementation would abstract this concept. For example, the Debian package rebuild procedure is fairly involved [1]. The procedure has to make sure we have deb-src repos for the debian package one wants to build. Also, it has to ensure the dpkg-dev package is installed, so on and so forth.

About the use-case, is this to support gcc/gdb test suite run in Red Hat based distributions? This is the closest I can think of where this feature could be useful.

So, to summarize, if that's something that is going to benefit users wanting to run tests in distribution package sources, I see no problem whatsoever. Just keep in mind it might not be very straightforward to implement the concept for RPM/DPKG backends.

Also, I would suggest to slightly change the method spec:

Input: package_name
Output: ready_to_build_path (children of self.srcdir)

[1] https://raphaelhertzog.com/2010/12/15/howto-to-rebuild-debian-packages/

**Comment by harish-24** [comment]
@lmr About the question, Yes it is mainly targeted to run gcc, gdb etc test suites. Thanks.

**Comment by ldoktor** [comment]
Correct me if I'm wrong, but this support was already included, wasn't it? Feel free to re-open if some part is still missing.
